### PR TITLE
Use linters' analyze from build command (CRAFT-318)

### PR DESCRIPTION
**Pull request by facundobatista**
Call linters' analyze from build command and deal with the results: so far only log the attribute ones (in the future it will deal also with warnings and errors), and use them for the manifest creation. Also changed the manifest creation function to properly save this new info.

Note: I had to fix checkers/analyze to save information in the shared state (a checkers integration test was missing).

Note 2: This PR depends on #443.

### CODE REVIEW

**Comment by facundobatista** [comment]
Also added some names fixing (a leftover from previous branch).

File: charmcraft/commands/build.py [hunk 1]
```diff
@@ -164,13 +165,33 @@ def build_charm(self, bases_config: BasesConfiguration) -> str:
...
 
-        create_manifest(self.buildpath, self.config.project.started_at, bases_config)
-
         linked_entrypoint = self.handle_generic_paths()
         self.handle_dispatcher(linked_entrypoint)
         self.handle_dependencies()
-        zipname = self.handle_package(bases_config)
 
+        linting_results = []
+        # run linters, present them to the user according to their type, and decide if go on
```
Comment (sergiusens) [inline charmcraft/commands/build.py:173]: "decide if go on" is missing words, I would inverse the statement to not use so many words
```suggestion
        # run linters, present them to the user according to their type, and fail if necessary
```

or maybe just remove as there is not decision being made yet.

**Review by sergiusens — APPROVED** [review]

File: charmcraft/commands/build.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/build.py]: thanks, improved

Comment (facundobatista) [reply charmcraft/commands/build.py]: btw, left it, as I put the XXX below...

**Review by cmatsuoka — APPROVED** [review]
Looks good (and it doesn't seem to conflict with parts support integration).

**Review by cjp256 — APPROVED** [review]
LGTM, but would suggest moving the linter logic into its own method(s).

**Pull request merged**